src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
+import { useEffect, useState } from 'react';
+import { getFutureDate } from '../utils';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listToken, itemId, data }) {
+	const [checkedState, setCheckedState] = useState(data.checked);
+
+	const checkTimePast = () => {
+		if (data.dateLastPurchased !== null) {
+			const yesterday = getFutureDate(-1);
+			const dateLastPurchased = new Date(data.dateLastPurchased.toDate());
+
+			return yesterday > dateLastPurchased;
+		}
+
+		return true;
+	};
+
+	const handleChange = () => {
+		if (checkTimePast() || data.checked === false) {
+			setCheckedState(!data.checked);
+			updateItem(listToken, itemId, !data.checked);
+		}
+	};
+
+	useEffect(function unCheckAfterTime() {
+		if (checkTimePast() && data.checked === true) {
+			setCheckedState(!data.checked);
+			updateItem(listToken, itemId, !data.checked);
+		}
```

Comment:
There is commonality that could be extracted into a common function, but probably not worth it.

Reply:
but I dunno, you might find it fun to do?